ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs

```diff
@@ -342,7 +343,10 @@ instance MaxTxSizeUTxO (ShelleyEra c) where
       SL.ApplyTxError . pure
     $ ShelleyEra.UtxowFailure
     $ ShelleyEra.UtxoFailure
-    $ ShelleyEra.MaxTxSizeUTxO x y
+    $ ShelleyEra.MaxTxSizeUTxO
+    -- TODO is this in the right order?
```

Comment:
It's called like this `validateMaybe (maxTxSizeUTxO txsz limit) $ ...`

So, `y` is the "expected" and `x` is the "supplied".

✔️ --- so please remove this TODO (and maybe rename `x` and `y` and add some Haddock on `maxTxSizeUTxO` now that the constructor is using record syntax).

Reply:
Maybe there's not much we could do, but I find the use of the `Mismatch` constructor misleading: It's not that we expected `limit` but got `txsz`; instead, we exceeded a specific threshold value.

Reply:
I agree. But that does seem like Ledger's decision, not ours.

Reply:
Another out-of-scope comment: shouldn't `MaxTxSizeUTxO` be called something like `MaxTxSizeUTxOError`?

Reply:
If you look at the type of `Mismatch` it has a phantom type that indicates the less then equal relation:
```haskell
  | MaxTxSizeUTxO
      !(Mismatch 'RelLTEQ Integer)
```

https://github.com/IntersectMBO/cardano-ledger/blob/cea57499d22c8acfce2a6b2d3c5c7ba59caa147e/eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Utxo.hs#L177-L178

So, with `Mismatch` we can now easily infer just by looking at the predicate type definition of what is the current vs expected value and their expected relationship.

> Another out-of-scope comment: shouldn't MaxTxSizeUTxO be called something like MaxTxSizeUTxOError?

It is not an *Error*, it is a predicate failure. Historically we never added *PredicateFailure* to every constructor in every predicate failure. So, no, IMHO it should not be called that.